timemachine/fe/rbfe.py

```diff
@@ -1,102 +1,109 @@
-import functools
-import io
 import pickle
 import traceback
 import warnings
-from typing import Any, Dict, NamedTuple, Optional, Sequence
+from typing import Any, Dict, List, NamedTuple, Optional, Sequence
 
-import matplotlib.pyplot as plt
 import numpy as np
-import pymbar
 from rdkit import Chem
 from simtk.openmm import app
 
-from timemachine.constants import BOLTZ, DEFAULT_TEMP
+from timemachine.constants import BOLTZ, DEFAULT_PRESSURE, DEFAULT_TEMP
 from timemachine.fe import atom_mapping, model_utils
-from timemachine.fe.bar import bar_with_bootstrapped_uncertainty
-from timemachine.fe.free_energy import HostConfig, InitialState, SimulationProtocol, SimulationResult
+from timemachine.fe.bar import bar_with_bootstrapped_uncertainty, df_err_from_ukln, pair_overlap_from_ukln
+from timemachine.fe.energy_decomposition import EnergyDecomposedState, compute_energy_decomposed_u_kln, get_batch_u_fns
+from timemachine.fe.free_energy import HostConfig, InitialState, SimulationProtocol, SimulationResult, sample
+from timemachine.fe.plots import make_dG_errs_figure, make_overlap_detail_figure, make_overlap_summary_figure
 from timemachine.fe.single_topology import SingleTopology
 from timemachine.fe.system import convert_omm_system
 from timemachine.fe.utils import get_mol_name, get_romol_conf
 from timemachine.ff import Forcefield
-from timemachine.lib import LangevinIntegrator, MonteCarloBarostat, custom_ops
+from timemachine.lib import LangevinIntegrator, MonteCarloBarostat
 from timemachine.md import builders, minimizer
 from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.parallel.client import AbstractClient, AbstractFileClient, CUDAPoolClient, FileClient
 from timemachine.potentials import jax_utils
 
 
-def get_batch_U_fns(bps):
-    # return a function that takes in coords, boxes
-    all_U_fns = []
-    for bp in bps:
+def setup_host(st: SingleTopology, host_config: HostConfig):
 
-        def batch_U_fn(xs, boxes, bp_impl):
-            Us = []
-            for x, box in zip(xs, boxes):
-                # tbd optimize to "selective" later
-                _, U = bp_impl.execute(x, box)
-                Us.append(U)
-            return np.array(Us)
+    host_system, host_masses = convert_omm_system(host_config.omm_system)
+    host_conf = minimizer.minimize_host_4d(
+        [st.mol_a, st.mol_b],
+        host_config.omm_system,
+        host_config.conf,
+        st.ff,
+        host_config.box,
+    )
+    host = (host_system, host_masses, host_conf)
```

Comment:
nit: this style is a bit odd when compared to `setup_in_vacuum`

Reply:
Possibly addressed in https://github.com/proteneer/timemachine/pull/936/commits/5df188cdebdf0395a58ce27245b69a14488daacd